src/main/java/com/jgcolab/mymenu/api/MenuController.java

```diff
@@ -62,7 +67,8 @@ public Menu registerMenu(@RequestBody Menu menu) {
 		}
 	}
 
-	@DeleteMapping ("/menu/{id}")
+	@ApiOperation("Delete a menu by ID.")
+	@DeleteMapping (value = "/menu/{id}", produces = "application/json")
 	@Transactional
 	public ResponseEntity<Menu> removeById(@PathVariable Long id) {
 		Optional<Menu> menu = menuRepository.findById(id);
```

Comment:
you could do everyting in a single line like:

menuRepository.findById(id).findFirst().map(c -> ReponseEntity()).orElse(sometingElse)